crates/polars-time/src/chunkedarray/string/strptime.rs

```diff
@@ -23,8 +23,11 @@ fn update_and_parse<T: atoi_simd::Parse>(
     let new_offset = offset + incr;
     let bytes = vals.get(offset..new_offset)?;
     let (val, parsed) = atoi_simd::parse_any(bytes).ok()?;
-    if parsed == 0 {
-        None
+    if parsed != incr {
+        panic!(
+            "Invariant when calling StrpTimeState.parse was not upheld. Expected {} parsed digits, got {}.",
+            incr, parsed
+        );
```

Comment:
is it ok to panic here? if we get here, it's a sign that something is seriously wrong, and it would've managed to avoid the bug pointed out in https://github.com/pola-rs/polars/issues/22167#issuecomment-2794781438

```python
In [21]: pl.Series(['2025-04-06T18:57:42.77756192Z']).str.to_datetime('%Y-%m-%dT%H:%M:%S.%9fcabbagebananapotato')
Out[21]:
shape: (1,)
Series: '' [datetime[ns]]
[
        2025-04-06 18:57:42.077756192
]
```